pkgs/tools/networking/linux-router/default.nix

```diff
@@ -0,0 +1,102 @@
+# SOURCE: https://github.com/NixOS/nixpkgs/pull/92787/files
+
+{ stdenv, lib, pkgs, fetchFromGitHub
+
+# Dependencies
+, bash , procps , iproute2 , dnsmasq , iptables
+
+# WiFi hotspot dependencies
+, hostapd , iw
+, wirelesstools ? null # for iwconfig
+, haveged ? null
+, qrencode ? null
+
+# General dependencies
+, coreutils , flock , gawk , getopt , gnugrep , gnused , which
+, networkmanager ? null
+
+# To fall back to haveged if entropy is low.
+#
+# Defaulting to false because not having it does not break things.
+# If it is really needed, warnings will be logged to journal.
+, useHaveged ? false
+
+# nmcli is not required for create_ap.
+# (defaulting to true because it is very likely already present)
+, useNetworkManager ? true
+
+# You only need this if 'iw' can not recognize your adapter.
+, useWirelessTools ? true
+
+# You only need this if you wish to show WiFi QR codes in terminal
+, useQrencode ? true
+}:
+
+assert useHaveged        -> !isNull haveged;
+assert useNetworkManager -> !isNull networkmanager;
+assert useWirelessTools  -> !isNull wirelesstools;
+assert useQrencode       -> !isNull qrencode;
+
+with lib;
+
+let
```

Comment:
This `let` can be moved directly to install phase as `binPath` is only used there. e.g.:
```
installPhase = let 
 ....
in ''
```

Reply:
I changed it, thanks :)